business_requirement/models/business.py

```diff
@@ -223,23 +225,77 @@ def project_id_change(self):
+                for project in project_id.message_follower_ids:
+                    msg_vals = {
+                        'channel_id': project.channel_id.id,
+                        'display_name': project.display_name,
+                        'partner_id': project.partner_id.id,
+                        'res_model': self._name,
+                        'subtype_ids': project.subtype_ids.ids
+                    }
+                    msg_followers.append((0, 0, msg_vals))
+                if msg_followers:
+                    vals['message_follower_ids'] = msg_followers
         return super(BusinessRequirement, self).create(vals)
 
     @api.multi
     def write(self, vals):
         if vals.get('project_id'):
-            project_id = self.env['project.project'].\
+            project_id = self.env['project.project']. \
                 browse(vals.get('project_id'))
             if project_id and project_id.message_follower_ids:
-                vals['message_follower_ids'] =\
-                    project_id.message_follower_ids.ids
+                msg_followers = []
+                for followers in self.message_follower_ids:
+                    msg_followers.append((2, followers.id))
+                for project in project_id.message_follower_ids:
+                    msg_vals = {
+                        'channel_id': project.channel_id.id,
+                        'display_name': project.display_name,
+                        'partner_id': project.partner_id.id,
+                        'res_model': self._name,
+                        'subtype_ids': project.subtype_ids.ids
+                    }
+                    msg_followers.append((0, 0, msg_vals))
+                if msg_followers:
+                    vals['message_follower_ids'] = msg_followers
+        if vals.get('state'):
+            ir_obj = self.env['ir.model.data']
+            br_xml_id = ir_obj.\
+                get_object('business_requirement',
+                           'group_business_requirement_manager')
+            user = self.env['res.users']
+            grps = [grp.id for grp in user.browse(self._uid).groups_id]
+            date = fields.Datetime.now()
+            if vals['state'] == 'confirmed':
+                vals.update({'confirmed_id': user,
+                             'confirmation_date': date})
+            if vals['state'] == 'draft':
+                vals.update({'confirmed_id': False,
+                             'approved_id': False,
+                             'confirmation_date': False,
+                             'approval_date': False
+                             })
+            if vals['state'] == 'approved':
+                if br_xml_id.id in grps:
+                    vals.update({'approved_id': user,
+                                 'approval_date': date})
+                else:
+                    raise ValidationError(_(
+                        'You can only move to the following stage:'
```

Comment:
If I'm correct, there wont be any space between `:` and `draft` in the message. Please add a space (same for other error messages)